metals/src/main/scala/scala/meta/internal/builds/BuildTools.scala

```diff
@@ -81,6 +81,9 @@ final class BuildTools(
     _.resolve("build.sc").isFile
   )
   def isMill: Boolean = millProject.isDefined
+  def isMillBsp(path: AbsolutePath): Boolean =
+    isInBsp(path) && path.filename == "mill-bsp.json"
```

Comment:
```suggestion
    isInBsp(path) && path.filename.contains("mill") && path.isJson
```
maybe more future proof.

Reply:
What if someone would have their custom BSP called `mill-fork.json`? Maybe exact match is better for detecting custom BSPs

Reply:
It's unlikely, while change of name might happen easily. Someone might come up with an idea to do just `mill.json` etc.